prow/gerrit/adapter/adapter.go

```diff
@@ -419,14 +419,40 @@ func (c *Controller) handleInRepoConfigError(err error, instance string, change
 }
 
 // processChange creates new presubmit/postsubmit prowjobs base off the gerrit changes
-func (c *Controller) processChange(logger logrus.FieldLogger, instance string, change client.ChangeInfo) error {
+func (c *Controller) processChange(logger logrus.FieldLogger, instance string, change client.ChangeInfo, lastUpdated time.Time) error {
 	tracker := time.Now()
 	defer func() {
 		// tracker will reset in `processChange` function, and this is the time
 		// taken for a gerrit change to be processed after the last reset.
 		logger.WithField("duration(s)", time.Since(tracker).Seconds()).Debug("Time taken for gerrit change to be processed.")
 	}()
 
+	if change.Revisions != nil {
+		var hasTestOrRetestCommand bool
+		for _, message := range currentMessages(change, lastUpdated) {
+			if strings.HasPrefix(message.Message, "/test") || strings.HasPrefix(message.Message, "/retest") {
+				hasTestOrRetestCommand = true
+				break
+			}
+		}
+
+		var hasCodeChange bool
+		for _, rev := range change.Revisions {
+			if rev.Created.Time.After(lastUpdated) && rev.Kind != gerrit.NoCodeChange {
+				hasCodeChange = true
+				break
+			}
+		}
+
+		if !hasTestOrRetestCommand && !hasCodeChange {
```

Comment:
I think this will skip postsubmit jobs, might need to move this entire closure to be under `client.New`

Reply:
Moved the whole closure to `client.New`.